api/serialization/storage.go

```diff
@@ -82,6 +82,31 @@ func (d *RawDataset) AddField(variable *model.Variable) error {
 	return nil
 }
 
+// InsertField inserts a field in the dataset at the specified index, updating both the data
+// and the metadata.
+func (d *RawDataset) InsertField(variable *model.Variable, index int) error {
```

Comment:
We may want to make this function increment the index for all variables at index+1...len(variables)